### PR TITLE
refactor(daemon): Index formulas by number

### PR DESCRIPTION
**Pull request by kriskowal**
This is a sequence of refactors that get us closer to indexing controllers by formula numbers (without type).

### CODE REVIEW

File: packages/daemon/src/formula-identifier.js [hunk 1]
```diff
@@ -1,24 +1,68 @@
+/// <ref types="ses">
+
 const { quote: q } = assert;
 
+const idPattern =
+  /^(?:(?<type>endo|readable-blob|worker|pet-store|pet-inspector|eval|lookup|make-unconfined|make-bundle|host|guest|handle|peer|directory|least-authority|loopback-network):)?(?<number>[0-9a-f]{128}):(?<node>[0-9a-f]{128})|(?<type2>web-bundle):(?<number2>[0-9a-f]{32})(?<node2>:[0-9a-f]{128})$/;
```
Comment (rekmarks) [inline packages/daemon/src/formula-identifier.js:6]: The last `web-bundle` capture group includes the final `:`.
```suggestion
const idPattern =
  /^(?:(?<type>endo|readable-blob|worker|pet-store|pet-inspector|eval|lookup|make-unconfined|make-bundle|host|guest|handle|peer|directory|least-authority|loopback-network):)?(?<number>[0-9a-f]{128}):(?<node>[0-9a-f]{128})|(?<type2>web-bundle):(?<number2>[0-9a-f]{32}):(?<node2>[0-9a-f]{128})$/;
```

File: packages/daemon/src/formula-identifier.js [hunk 2]
```diff
@@ -1,24 +1,68 @@
...
+
+/**
+ * @param {string} formulaIdentifier
```
Comment (rekmarks) [inline packages/daemon/src/formula-identifier.js:9]: Would we welcome a PR of `s/formulaIDentifier/formulaId/g` after this is merged?

File: packages/daemon/src/formula-identifier.js [hunk 3]
```diff
@@ -1,24 +1,68 @@
...
-  `${type}:${number}:${node}`;
+export const formatId = ({ type, number, node }) => {
+  const parts = [number];
+  if (type !== undefined) {
+    parts.unshift(type);
+  }
+  if (node !== undefined) {
+    parts.push(node);
+  }
+  return parts.join(':');
```
Comment (rekmarks) [inline packages/daemon/src/formula-identifier.js:67]: As a defensive measure, should we call `assertValidId` on the constructed id before returning it? If so, we should perhaps change the error message to differentiate this case from that of an invalid already-constructed id.

```suggestion
  const id = parts.join(':');
  assertValidId(id);
  return id;
```

**Review by rekmarks — CHANGES_REQUESTED** [review]
One RegEx-related bug, then just optional questions and suggestions.

File: packages/daemon/src/formula-identifier.js (hunk 2 above)
Comment (kriskowal) [reply packages/daemon/src/formula-identifier.js:9]: I would like to do this and we should book a quiet day to do it on.

File: packages/daemon/src/daemon.js [hunk 4]
```diff
@@ -698,24 +708,19 @@ const makeDaemonCore = async (
...
-      );
-    } else {
-      throw new TypeError(
-        `Invalid formula identifier, unrecognized type ${q(formulaIdentifier)}`,
-      );
+      assert.Fail`Invalid formula identifier, unrecognized type ${q(
+        allegedFormulaIdentifier,
+      )}`;
     }
+    // TODO further validation
```
[the remark this answers is not in the document]
Comment (kriskowal) [reply packages/daemon/src/daemon.js:717]: We would presumably validate the shape farther down, in each individual formula type reviver. For the moment, we assume an absence of corruption, either automatic or manual. We can probably get away with that for a while.

File: packages/daemon/src/pet-store.js [hunk 5]
```diff
@@ -248,7 +238,7 @@ export const makePetStoreMaker = (filePowers, locator) => {
    * @returns {Promise<import('./types.js').PetStore>}
    */
   const makeIdentifiedPetStore = (id, assertValidName) => {
-    if (!validIdPattern.test(id)) {
+    if (!validNumberPattern.test(id)) {
```
Comment (rekmarks) [inline packages/daemon/src/pet-store.js:241]: The `id` param here should have been renamed `formulaNumber` in a previous PR. Likewise in `types.d.ts`.

**Review by rekmarks — APPROVED** [review]
One suggestion, otherwise LGTM!

**Pull request merged**